app/controllers/tournaments_controller.rb

```diff
@@ -20,6 +40,13 @@ def scores
   end
 
   def success
+    begin
+      results = Result.select(:answers).where(tournament_id: params[:id])
+      statistics = StatisticsCalculator.success_per_question(results)
+      render json: statistics, status: :ok 
+    rescue
+      render json: { error: "Internal Server Error" }, status: :internal_server_error
```

Comment:
the begin and rescue here don't contribute at the moment, since rails will render 
`      render json: { error: "Internal Server Error" }, status: :internal_server_error
`automatic in case of uncaught exception. common handlings can be:
1. rescue Exception => e
Rails.logger.error("some text #{e.message}")

2. Check after action weather the succeed, e:g (the code snippet is assume that Result model must has tournament id):
```
results = Result.find_by(tournament_id: params[:id])
render json: { error: "Tournament with id #{params[:id]} not found" }, status: :not_found and return if results.blank?

```

* Rails best practice use '' (x = 'str') for static string and "" for dynamic (y = 10 x = "y value is #{y}")